src/vs/platform/windows/electron-main/windowImpl.ts

```diff
@@ -707,6 +743,23 @@ export class CodeWindow extends Disposable implements ICodeWindow {
 					let message: string;
 					if (!details) {
 						message = localize('appGone', "The window terminated unexpectedly");
+					} else if (details.reason === 'launch-failed' && details.exitCode === 18) {
+						// Show Dialog
+						const { response } = await this.dialogMainService.showMessageBox({
+							type: 'warning',
+							buttons: [
+								localize({ key: 'restart', comment: ['&& denotes a mnemonic'] }, "&&Restart")
+							],
+							message: localize('appSandboxError', "Chromium sandbox is unable to create workbench process in your environment"),
+							detail: localize('appSandboxErrorDetail', "We are sorry for the inconvenience. You can restart the application with chromium sandbox disabled to continue where you left off.")
+						}, this._win);
+						if (response === 0) {
+							this.updateChromiumSandboxEnablement(false);
+							this.lifecycleMainService.relaunch();
```

Comment:
I think we should still destroy the window to ensure an orderly shutdown. I would suggest to have another parameter for `destroyWindow` to trigger a relaunch instead of a reload of the window.